tasks/kmt.py

```diff
@@ -58,6 +60,59 @@ def gen_config(
     vmconfig.gen_config(ctx, stack, vms, sets, init_stack, vcpu, memory, new, ci, arch, output_file)
 
 
+@task(
+    help={
+        "pipeline": "Pipeline ID to use",
+        "stack": "Name of the stack within which to generate the configuration file",
+        "vcpu": "Comma separated list of CPUs, to launch each VM with",
+        "memory": "Comma separated list of memory to launch each VM with. Automatically rounded up to power of 2",
+        "new": "Generate new configuration file instead of appending to existing one within the provided stack",
+        "init-stack": "Automatically initialize stack if not present. Equivalent to calling 'inv -e kmt.create-stack [--stack=<stack>]'",
+    }
+)
+def gen_config_from_ci_pipeline(
+    ctx,
+    stack=None,
+    pipeline=None,
+    init_stack=False,
+    vcpu="4",
+    memory="8192",
```

Comment:
If the gitlab library allows us to get artifacts, then the KMT configuration file is exposed as an artifact for the [kernel_matrix_testing_setup_env_* jobs](https://github.com/DataDog/datadog-agent/blob/main/.gitlab/kernel_version_testing/system_probe.yml#L320).

Maybe we can use this to set vcpu and memory. So that the VM configuration also exactly matches the CI.

Reply:
Done! I've used the first ocurrence for simplicity, from what I've seen all configurations use the same.

Reply:
It is technically possible for configurations to have different ones, but I don't think that would ever happen in practice so this is good enough.